### PR TITLE
[PPP-4660] - Use of Vulnerable Component: Components/jackson-databind:2.9.10.2

### PR DESCRIPTION
**Pull request by andreramos89**
@smmribeiro 
@renato-s 
@bcostahitachivantara

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 2m 36s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/common-resources,assemblies/server
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 59s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/common-resources,assemblies/server
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: assemblies/common-resources/src/main/resources-filtered/etc/org.apache.karaf.features.xml [hunk 1]
```diff
@@ -60,11 +60,11 @@
 
         <!-- [PPP-4108] Use of vulnerable component jackson-mapper-asl-1.9.2.jar CVE-2017-7525, CVE-2017-15095, CVE-2017-15095 -->
         <bundle
-            originalUri="mvn:org.codehaus.jackson/jackson-core-asl/[1.5.0,${codehaus-jackson.version})"
-            replacement="mvn:org.codehaus.jackson/jackson-core-asl/${codehaus-jackson.version}" mode="maven" />
+            originalUri="mvn:org.codehaus.jackson/jackson-core-asl/[1.5.0,1.9.14)"
```
Comment (smmribeiro) [inline assemblies/common-resources/src/main/resources-filtered/etc/org.apache.karaf.features.xml:63]: We should be replacing every version until "${fasterxml-jackson}", not "1.9.14"!
So it should be like this:
`          originalUri="mvn:org.codehaus.jackson/jackson-core-asl/[1.5.0,${fasterxml-jackson})"`

If one leaves the code as is, a jar whose version falls in the interval between 1.9.14 and 2.14.1 (current value for ${fasterxml-jackson}) would not be replaced, which could bring some inconsistencies.

File: assemblies/common-resources/src/main/resources-filtered/etc/org.apache.karaf.features.xml [hunk 2]
```diff
@@ -60,11 +60,11 @@
...
+            replacement="mvn:com.fasterxml.jackson.core/jackson-core/${fasterxml-jackson}" mode="maven" />
         <bundle
-            originalUri="mvn:org.codehaus.jackson/jackson-mapper-asl/[1.5.0,${codehaus-jackson.version})"
-            replacement="mvn:org.codehaus.jackson/jackson-mapper-asl/${codehaus-jackson.version}" mode="maven" />
+            originalUri="mvn:org.codehaus.jackson/jackson-mapper-asl/[1.5.0,1.9.14)"
```
Comment (smmribeiro) [inline assemblies/common-resources/src/main/resources-filtered/etc/org.apache.karaf.features.xml:66]: Similar case to the previous comment.

File: assemblies/server/src/main/resources-filtered/etc/org.apache.karaf.features.xml [hunk 3]
```diff
@@ -61,11 +61,11 @@
 
         <!-- [PPP-4108] Use of vulnerable component jackson-mapper-asl-1.9.2.jar CVE-2017-7525, CVE-2017-15095, CVE-2017-15095 -->
         <bundle
-            originalUri="mvn:org.codehaus.jackson/jackson-core-asl/[1.5.0,${codehaus-jackson.version})"
-            replacement="mvn:org.codehaus.jackson/jackson-core-asl/${codehaus-jackson.version}" mode="maven" />
+                originalUri="mvn:org.codehaus.jackson/jackson-core-asl/[1.5.0,1.9.14)"
```
Comment (smmribeiro) [inline assemblies/server/src/main/resources-filtered/etc/org.apache.karaf.features.xml:64]: Similar case to the previous comments.

File: assemblies/server/src/main/resources-filtered/etc/org.apache.karaf.features.xml [hunk 4]
```diff
@@ -61,11 +61,11 @@
...
+                replacement="mvn:com.fasterxml.jackson.core/jackson-core/${fasterxml-jackson}" mode="maven" />
         <bundle
-            originalUri="mvn:org.codehaus.jackson/jackson-mapper-asl/[1.5.0,${codehaus-jackson.version})"
-            replacement="mvn:org.codehaus.jackson/jackson-mapper-asl/${codehaus-jackson.version}" mode="maven" />
+                originalUri="mvn:org.codehaus.jackson/jackson-mapper-asl/[1.5.0,1.9.14)"
```
Comment (smmribeiro) [inline assemblies/server/src/main/resources-filtered/etc/org.apache.karaf.features.xml:67]: Similar case to the previous comments.

File: assemblies/common-resources/src/main/resources-filtered/etc/org.apache.karaf.features.xml (hunk 2 above)
Comment (andreramos89) [reply assemblies/common-resources/src/main/resources-filtered/etc/org.apache.karaf.features.xml:66]: https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-mapper-asl - org.codehaus.jackson/jackson-mapper-asl was moved to com.fasterxml.jackson.core/jackson-databind

File: assemblies/server/src/main/resources-filtered/etc/org.apache.karaf.features.xml (hunk 3 above)
Comment (andreramos89) [reply assemblies/server/src/main/resources-filtered/etc/org.apache.karaf.features.xml:64]: https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-core-asl - the org.codehaus.jackson/jackson-core-asl was moved to com.fasterxml.jackson.core/jackson-core

File: assemblies/server/src/main/resources-filtered/etc/org.apache.karaf.features.xml (hunk 4 above)
Comment (andreramos89) [reply assemblies/server/src/main/resources-filtered/etc/org.apache.karaf.features.xml:67]: https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-mapper-asl - org.codehaus.jackson/jackson-mapper-asl was moved to com.fasterxml.jackson.core/jackson-databind

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 51s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/common-resources,assemblies/server
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 52s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/common-resources,assemblies/server
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 59s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/common-resources,assemblies/server
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by smmribeiro** [comment]
Closing this and all "January 2023" related PRs for this issue (check this [comment](https://hv-eng.atlassian.net/browse/PPP-4660?focusedCommentId=1546417) and this [comment](https://hv-eng.atlassian.net/browse/PPP-4660?focusedCommentId=1543281)).

Too much time has passed and the risk of a PR being incomplete (or simply wrong) is too high.
I won't delete the original branch so that anyone picking up the development can see what was done previously.

**Pull request closed without merging**